daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+
+        return ev_list_path
+    
+    def download(self, num_cores: int = NUM_CORES):
+        """
+        A method to acquire and download the eROSITA CalPV data that have not been filtered out (if a filter
+        has been applied, otherwise all data will be downloaded). Fields (or field types), which is set either 
+        on declaration of the class instance or by passing a new value to the chosen_instruments property.
+        """
+        # Ensures that a directory to store the 'raw' pointed XMM data in exists - once downloaded and unpacked
+        #  this data will be processed into a DAXA 'archive' and stored elsewhere.
+        if not os.path.exists(self.top_level_path + self.name + '_raw'):
+            os.makedirs(self.top_level_path + self.name + '_raw')
+        # Grabs the raw data storage path
+        stor_dir = self.raw_data_path
+
+        # A very unsophisticated way of checking whether raw data have been downloaded before (see issue #30)
+        #  If not all data have been downloaded there are also secondary checks on an ObsID by ObsID basis in
+        #  the _download_call method
+        if all([os.path.exists(stor_dir + '{o}'.format(o=o)) for o in self.filtered_obs_ids]):
+            self._download_done = True
+        
+        # Since multiple Obs_IDs are downloaded with one field, we are first downloading all fields needed.
+        # This dictionary contains which obs_ids are associated with which field
+        field_dict = self._field_dict_generator()
+        # A list of fields needed to get the required obs_ids (with no duplicate fields)
+        required_fields = list(set([field_dict[o] for o in self.filtered_obs_ids]))
+
+        # Dictionary containing all the possible obs_ids associated with one field (field keys, obs_ids values as a list)
+        obs_dict = {}
+        for field in required_fields:
+            obs_dict[field] = [obs for obs in field_dict if field_dict[obs] == field]
+        
+        # Collect all the fields who have got an obs_id/ obs_ids that aren't already downloaded 
+        fields_to_be_downloaded = [field for field in required_fields if not all([os.path.exists(stor_dir + '{o}'.format(o=o)) for o in obs_dict[field]])]
+        
+        if not self._download_done:
+            # If only one core is to be used, then it's simply a case of a nested loop through ObsIDs and instruments
+            if num_cores == 1:
+                with tqdm(total=len(fields_to_be_downloaded), desc="Downloading {} data".format(self._pretty_miss_name)) as download_prog:
+                    for field in fields_to_be_downloaded:
+                        # Use the internal static method I set up which both downloads and unpacks the eROSITACalPV data
+                        self._download_call(self, field=field)
+                        # Update the progress bar
+                        download_prog.update(1)
+
+            elif num_cores > 1:
```

Comment:
I'm afraid this chunk just fully doesn't work - it gives errors about there being no self passed to the _download_call method. That will go away once you make it static again and then as far as I can tell it will work